https://github.com/Kitware/CMake/commit/c92268f91b30e6d52b82d1ffad227cadafcf4dbf
From: Matt Jolly <kangie@gentoo.org>
Date: Thu, 11 Sep 2025 11:20:42 +1000
Subject: [PATCH] cmCTestCurl: Avoid using undocumented type for
 CURLOPT_PROXYTYPE values

Since upstream curl commit `1a12663d06` (CURLOPT: bump `CURLPROXY_*`
enums to `long`, drop casts, 2025-07-28), the `CURLPROXY_*` constants
are integer literals instead of `enum curl_proxytype`.  It turns out
that `curl_easy_setopt` has always expected a `long` anyway, and that
`curl_proxytype` is not documented for public use.

Rebased for 3.28.5

Signed-off-by: Matt Jolly <kangie@gentoo.org>
--- a/Source/CTest/cmCTestCurl.h
+++ b/Source/CTest/cmCTestCurl.h
@@ -44,7 +44,7 @@ private:
   std::vector<std::string> HttpHeaders;
   std::string HTTPProxyAuth;
   std::string HTTPProxy;
-  curl_proxytype HTTPProxyType;
+  long HTTPProxyType;
   bool VerifyHostOff;
   bool VerifyPeerOff;
   bool UseHttp10;
-- 
2.49.1
